core/src/main/java/de/danoeh/antennapod/core/preferences/UserPreferences.java

```diff
@@ -797,29 +806,51 @@ public static void restartUpdateAlarm(boolean now) {
      */
     private static void restartUpdateIntervalAlarm(long triggerAtMillis, long intervalMillis) {
         Log.d(TAG, "Restarting update alarm.");
+
+        if (Build.VERSION.SDK_INT >= 24) {
+            JobScheduler jobScheduler = context.getSystemService(JobScheduler.class);
+            if (jobScheduler != null) {
+                JobInfo oldJob = jobScheduler.getPendingJob(JOB_ID_FEED_UPDATE);
+                if (oldJob == null || oldJob.getIntervalMillis() != intervalMillis) {
+                    JobInfo.Builder builder = getFeedUpdateJobBuilder();
+                    builder.setPeriodic(intervalMillis);
+                    jobScheduler.cancel(JOB_ID_FEED_UPDATE);
+
+                    if (intervalMillis <= 0) {
+                        Log.d(TAG, "Automatic update was deactivated");
+                        return;
+                    }
+
+                    jobScheduler.schedule(builder.build());
+                    Log.d(TAG, "JobScheduler was set at interval " + intervalMillis);
+                } else {
+                    Log.d(TAG, "JobScheduler was already set at interval " + intervalMillis + ", ignoring.");
+                }
+            }
+            return;
```

Comment:
this is surprising

I'd extract the new and old in two different methods, maybe even objects